rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -64,16 +65,15 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_topics_are
     static_cast<MockSequentialWriter &>(writer.get_implementation_handle());
 
   constexpr size_t expected_messages = 4;
-  auto ret = rosbag2_test_common::wait_until_shutdown(
-    std::chrono::seconds(5),
+  auto ret = rosbag2_test_common::wait_until_condition(
     [ =, &mock_writer]() {
       return mock_writer.get_messages().size() >= expected_messages;
-    });
+    },
+    std::chrono::seconds(5));
   auto recorded_messages = mock_writer.get_messages();
   EXPECT_TRUE(ret) << "failed to capture expected messages in time" <<
     "recorded messages = " << recorded_messages.size();
   EXPECT_THAT(recorded_messages, SizeIs(expected_messages));
-  stop_spinning();
```

Comment:
Suggestion to return deleted explict call to `stop_spinning()` and `cleanup_process_handle.cancel();`

Reply:
Same comment as https://github.com/ros2/rosbag2/pull/1796#discussion_r1752020122